Revised GUI interface and workflow

tnelson-integral opened the pull request:
Adds a tabbed GUI,
All modules can be run at a single time,
improved input names,
output files sorted by stressor module
QGIS layers sorted into stressor layer groups
package can be run without style files defined. 
New tutorial file structure is necessary (new link will be shared for testing purposes)

ssolson commented:
@tnelson-integral the Pacwave example is working except for the shear stress module I am getting the following error:
```
AttributeError: 'NoneType' object has no attribute 'GetRasterBand' 
Traceback (most recent call last):
  File "C:\Users/sterl/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\seat_qgis_plugin\stressor_receptor_calc.py", line 692, in run
    sfilenames = run_shear_stress_stressor(
  File "C:\Users/sterl/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\seat_qgis_plugin\modules\shear_stress_module.py", line 465, in run_shear_stress_stressor
    dict_of_arrays, rx, ry, dx, dy, gridtype = calculate_shear_stress_stressors(fpath_nodev=dev_notpresent_file,
  File "C:\Users/sterl/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\seat_qgis_plugin\modules\shear_stress_module.py", line 344, in calculate_shear_stress_stressors
    receptor_array = calc_receptor_array(
  File "C:\Users/sterl/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\seat_qgis_plugin\modules\stressor_utils.py", line 204, in calc_receptor_array
    img = data.GetRasterBand(1)
AttributeError: 'NoneType' object has no attribute 'GetRasterBand'
```

tnelson-integral commented:
I updated the tutorial file inputs to include the missing subfolder.
shear stress grain size file = <input_folder>/receptor/velocity_receptor.csv
shear stress risk layer file = <input_folder>/risk layer/habitat_classification.tif


Tim Nelson (he/him/his) | Project Scientist
Tel: 561.240.5096 | Cell: 803.319.3344
200 2nd Ave. S, #155 | St. Petersburg | FL 33701
***@***.*** | www.integral-corp.com

[cid:integral-logo_bb8ba854-3124-462b-8a66-06670ee4325c.jpg]

________________________________
From: ssolson ***@***.***>
Sent: Wednesday, December 20, 2023 11:38 AM
To: sandialabs/seat-qgis-plugin ***@***.***>
Cc: Tim Nelson ***@***.***>; Mention ***@***.***>
Subject: Re: [sandialabs/seat-qgis-plugin] Revised GUI interface and workflow (PR #31)

[CAUTION: External email. Think before you click links or open attachments.]

@tnelson-integral<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_tnelson-2Dintegral&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=2OT5Kb-Guh1vdHPYH_OKgQqqKZs-WcstcKgpghDlHN8&m=w7dUuySPjMDFonfMo6Y2xyuR4jhnP4Unap6NPDSfON_GiOYbMY5GUodBCz-_b1k-&s=SKg7iHAvNW0to1WPhvUEHK6Q32QHqjcO-62GruYEnpE&e=> the Pacwave example is working except for the shear stress module I am getting the following error:

AttributeError: 'NoneType' object has no attribute 'GetRasterBand'
Traceback (most recent call last):
  File "C:\Users/sterl/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\seat_qgis_plugin\stressor_receptor_calc.py", line 692, in run
    sfilenames = run_shear_stress_stressor(
  File "C:\Users/sterl/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\seat_qgis_plugin\modules\shear_stress_module.py", line 465, in run_shear_stress_stressor
    dict_of_arrays, rx, ry, dx, dy, gridtype = calculate_shear_stress_stressors(fpath_nodev=dev_notpresent_file,
  File "C:\Users/sterl/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\seat_qgis_plugin\modules\shear_stress_module.py", line 344, in calculate_shear_stress_stressors
    receptor_array = calc_receptor_array(
  File "C:\Users/sterl/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\seat_qgis_plugin\modules\stressor_utils.py", line 204, in calc_receptor_array
    img = data.GetRasterBand(1)
AttributeError: 'NoneType' object has no attribute 'GetRasterBand'


—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sandialabs_seat-2Dqgis-2Dplugin_pull_31-23issuecomment-2D1864794142&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=2OT5Kb-Guh1vdHPYH_OKgQqqKZs-WcstcKgpghDlHN8&m=w7dUuySPjMDFonfMo6Y2xyuR4jhnP4Unap6NPDSfON_GiOYbMY5GUodBCz-_b1k-&s=E-YwNuuAxPepOt5JdLhhDRTmcUk_8cbLa4JzdqN1Yqg&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AZARDBPWNFBX4U7LQIHIA7DYKMH6TAVCNFSM6AAAAABAXB2PBOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNRUG44TIMJUGI&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=2OT5Kb-Guh1vdHPYH_OKgQqqKZs-WcstcKgpghDlHN8&m=w7dUuySPjMDFonfMo6Y2xyuR4jhnP4Unap6NPDSfON_GiOYbMY5GUodBCz-_b1k-&s=OzSENfqkR9b6tw1Ofbsz4JdR4vDEZYg7YNwtHUMiqxY&e=>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

ssolson commented:
@tnelson-integral thanks for fixing the default file to address the error above. I think we should add a check to make sure the file exists and throw an error if it does not. This happens with a number of the other inputs but would help us have more meaningful errors.

tnelson-integral commented:
I think adding a check when importing an .ini file to see if the file exists is something that could be added with perhaps a status update in the GUI or turn the text red to provide feedback. I will look into the best methods within the GUI limits and add that to a future pull request.
-Tim

Tim Nelson (he/him/his) | Project Scientist
Tel: 561.240.5096 | Cell: 803.319.3344
200 2nd Ave. S, #155 | St. Petersburg | FL 33701
***@***.*** | www.integral-corp.com

[cid:integral-logo_bb8ba854-3124-462b-8a66-06670ee4325c.jpg]

________________________________
From: ssolson ***@***.***>
Sent: Wednesday, December 20, 2023 11:54 AM
To: sandialabs/seat-qgis-plugin ***@***.***>
Cc: Tim Nelson ***@***.***>; Mention ***@***.***>
Subject: Re: [sandialabs/seat-qgis-plugin] Revised GUI interface and workflow (PR #31)

[CAUTION: External email. Think before you click links or open attachments.]

@tnelson-integral<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_tnelson-2Dintegral&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=2OT5Kb-Guh1vdHPYH_OKgQqqKZs-WcstcKgpghDlHN8&m=F4q3zhLZWBnKJDMv19jdSUYSjxe0gK2tXIUy9pS7bool2_17eG9vNpfVaO_OYKhZ&s=cTtwMcX9MsEZ384LGd9kUDWt33LW7KcJQDeO8qhQH2c&e=> thanks for fixing the default file to address the error above. I think we should add a check to make sure the file exists and throw an error if it does not. This happens with a number of the other inputs but would help us have more meaningful errors.

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sandialabs_seat-2Dqgis-2Dplugin_pull_31-23issuecomment-2D1864817585&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=2OT5Kb-Guh1vdHPYH_OKgQqqKZs-WcstcKgpghDlHN8&m=F4q3zhLZWBnKJDMv19jdSUYSjxe0gK2tXIUy9pS7bool2_17eG9vNpfVaO_OYKhZ&s=z8sqXHkXUaUnBZHjREphRJwXuqSNiVtIEj6yVftWdkM&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AZARDBL3MTMUIHW4WSF6PZLYKMJ4NAVCNFSM6AAAAABAXB2PBOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNRUHAYTONJYGU&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=2OT5Kb-Guh1vdHPYH_OKgQqqKZs-WcstcKgpghDlHN8&m=F4q3zhLZWBnKJDMv19jdSUYSjxe0gK2tXIUy9pS7bool2_17eG9vNpfVaO_OYKhZ&s=hzDeDnDouF7oNODDa2GdyURTgM70EBgw0GkGkB_izA8&e=>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

ssolson commented:
> I think adding a check when importing an .ini file to see if the file exists is something that could be added with perhaps a status update in the GUI or turn the text red to provide feedback. I will look into the best methods within the GUI limits and add that to a future pull request.

While in the GUI would be nice I am talking about at the function level. It is as simple as adding the following:
```
if not os.path.exists(file_path):
        raise FileNotFoundError(f"The file {file_path} does not exist.")
```

tnelson-integral commented:
adding a check when importing an .ini file, check for files in module if run independently of GUI, added user feedback in GUI,m and updated toolbar icon.

ssolson commented:
Hey @tnelson-integral thanks for the update. I tried to run the latest version today. One issue I had was that your current PR needs to merge the main branch (e.g. there are conflicts). I accepted all incoming changes to run the modules. Could you merge the changes on this PR to fix the conflicts?

In running the PacWave example everything was working except the all_modules example. There seems to be ana issue importing the style files for this. Specifically the stack trace I get is:
```
An error has occurred while executing Python code: 

KeyError: "None of ['Type'] are in the columns" 
Traceback (most recent call last):
  File "C:\Users/sterl/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\seat_qgis_plugin\stressor_receptor_calc.py", line 713, in run
    stylefiles_DF = self.read_style_files(self.dlg.output_stylefile.text())
  File "C:\Users/sterl/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\seat_qgis_plugin\stressor_receptor_calc.py", line 246, in read_style_files
    data = data.set_index('Type')
  File "C:\PROGRA~1\QGIS33~1.1\apps\Python39\lib\site-packages\pandas\core\frame.py", line 5859, in set_index
    raise KeyError(f"None of {missing} are in the columns")
KeyError: "None of ['Type'] are in the columns"
```

This is strange since all the modules are using the same style files csv and the other example work. Could you take a look and let me know if this example is running for you?

tnelson-integral commented:
> Hey @tnelson-integral thanks for the update. I tried to run the latest version today. One issue I had was that your current PR needs to merge the main branch (e.g. there are conflicts). I accepted all incoming changes to run the modules. Could you merge the changes on this PR to fix the conflicts?
> 
> In running the PacWave example everything was working except the all_modules example. There seems to be an issue importing the style files for this. Specifically the stack trace I get is:
> 
> ```
> An error has occurred while executing Python code: 
> 
> KeyError: "None of ['Type'] are in the columns" 
> Traceback (most recent call last):
>   File "C:\Users/sterl/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\seat_qgis_plugin\stressor_receptor_calc.py", line 713, in run
>     stylefiles_DF = self.read_style_files(self.dlg.output_stylefile.text())
>   File "C:\Users/sterl/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\seat_qgis_plugin\stressor_receptor_calc.py", line 246, in read_style_files
>     data = data.set_index('Type')
>   File "C:\PROGRA~1\QGIS33~1.1\apps\Python39\lib\site-packages\pandas\core\frame.py", line 5859, in set_index
>     raise KeyError(f"None of {missing} are in the columns")
> KeyError: "None of ['Type'] are in the columns"
> ```
> 
> This is strange since all the modules are using the same style files csv and the other example work. Could you take a look and let me know if this example is running for you?

@ssolson 
There is a difference in the style_files_all_modules.default I was using locally and the one on OneDrive. Try downloading the updated version and see if that works.

ssolson commented:
@tnelson-integral there is a major bug introduced in the merge you pushed that prevents the user from being able to install seat. Once I fix that I am still getting errors on the new demo files.
```
KeyError: "None of ['Type'] are in the columns" 
Traceback (most recent call last):
  File "C:\Users/sterl/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\seat\stressor_receptor_calc.py", line 715, in run
    stylefiles_DF = self.read_style_files(self.dlg.output_stylefile.text())
  File "C:\Users/sterl/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\seat\stressor_receptor_calc.py", line 247, in read_style_files
    data = data.set_index('Type')
  File "C:\PROGRA~1\QGIS33~1.1\apps\Python39\lib\site-packages\pandas\core\frame.py", line 5859, in set_index
    raise KeyError(f"None of {missing} are in the columns")
KeyError: "None of ['Type'] are in the columns"
```

Could you please confirm that you can install the latest zip and run all the demo files as downloaded from the folder?

tnelson-integral commented:
@ssolson, I tracked down the error and the current plugin version is working on my end.

ssolson commented:
> @ssolson, I tracked down the error and the current plugin version is working on my end.

Thanks Tim. I was able to run all of the Tutorials. Heading into next week I will start the documentation refactor to go along with the updated Tabbed GUI and new demo files.

ssolson commented:
@tnelson-integral I have a couple questions and notes from trying to migrate the docs to the tabbed GUI format.

1. Is "Secondary Constraint" is now called "Species spatial Probability/ density directory"?
2. What does Temporal Avg do?
  - its says optional but it looks like it is always on. I assume it only does anything if the correct inputs are supplied but it is currently unclear. 
  - Also even if it only applies when the correct inputs are supplied it seems required as there is no clear "none" option. Maybe final timestep is this none that turns this off?
3. It seems the probabilities is now required. Is this correct?
4. There is a new field called the "Risk Layer". Could you describe this for the documentation?
The tooltips on Risk layer and Depth avg say they are for the secondary constraint.
![image](https://github.com/sandialabs/seat-qgis-plugin/assets/13438942/8ccf713b-4ae1-4b73-bb91-4671016eb531)

ssolson commented:
@tnelson-integral thank you for your updates to the documentation.

Can you help me understand why you are replacing some of these images?

E.g. in the png files you include `model_results_directory_input.png`. 
This is the same name and image as the webp version I had in the docs in commit 2a3ba61ec26ab1deb9436fde71d7602d86c79dff

When I include the version you sent me it renders poorly e.g. pixilated and small. I am just trying to understand if there is a reason you wanted to overwrite these when they appear the same to me. I was in the process of undoing this work and thought I should check.

Example Render from your png file
![image](https://github.com/sandialabs/seat-qgis-plugin/assets/13438942/e19929bd-531c-4db6-9504-45096a78355f)


My previous version from commit 2a3ba61ec26ab1deb9436fde71d7602d86c79dff
![image](https://github.com/sandialabs/seat-qgis-plugin/assets/13438942/bedb4726-8c85-471a-8101-4b1e885bbcb6)

ssolson commented:
Hey @tnelson-integral did you see https://github.com/IntegralEnvision/seat-qgis-plugin/pull/2 that I opened last week? Need to get those issues addressed to move this forward.

tnelson-integral commented:
Hi Sterling,
No, I did not get the notification via email. I will work on that this week. Thank you for following up.
-Tim

Tim Nelson (he/him/his) | Project Scientist
Tel: 561.240.5096 | Cell: 803.319.3344 | St. Petersburg | FL
***@***.*** | Bio<https://www.integral-corp.com/staff/timothy-r-nelson-ph-d>

<https://www.integral-corp.com/>[cid:integral_1a04a855-2877-4c4f-bd98-b89de3950d0f.png]<https://www.integral-corp.com/>

________________________________
From: ssolson ***@***.***>
Sent: Friday, March 15, 2024 10:52 AM
To: sandialabs/seat-qgis-plugin ***@***.***>
Cc: Tim Nelson ***@***.***>; Mention ***@***.***>
Subject: Re: [sandialabs/seat-qgis-plugin] Revised GUI interface and workflow (PR #31)

[CAUTION: External email. Think before you click links or open attachments.]

Hey @tnelson-integral<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_tnelson-2Dintegral&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=2OT5Kb-Guh1vdHPYH_OKgQqqKZs-WcstcKgpghDlHN8&m=JCSmwF3soIF5i5oJw54jFMkr1w7gJif6iGo27lOImSQI2PDWfh1mxi-m2T3NK7zu&s=R4POp9ZmgT25kfM7mE_8C0mrN9mfztgFaG9hnkGI7FM&e=> did you see IntegralEnvision#2<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_IntegralEnvision_seat-2Dqgis-2Dplugin_pull_2&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=2OT5Kb-Guh1vdHPYH_OKgQqqKZs-WcstcKgpghDlHN8&m=JCSmwF3soIF5i5oJw54jFMkr1w7gJif6iGo27lOImSQI2PDWfh1mxi-m2T3NK7zu&s=iCNX_TcEPKPksjsHJpq6HNQy8NsqjQK_W3Ywc-8eXK0&e=> that I opened last week? Need to get those issues addressed to move this forward.

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sandialabs_seat-2Dqgis-2Dplugin_pull_31-23issuecomment-2D1999836556&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=2OT5Kb-Guh1vdHPYH_OKgQqqKZs-WcstcKgpghDlHN8&m=JCSmwF3soIF5i5oJw54jFMkr1w7gJif6iGo27lOImSQI2PDWfh1mxi-m2T3NK7zu&s=HwnO6zVwjwmJpk8F9AaNnlSRsYo274gzxTVd0vhwOng&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AZARDBLBDPULYMCZQQSRYGDYYMDMFAVCNFSM6AAAAABAXB2PBOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOJZHAZTMNJVGY&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=2OT5Kb-Guh1vdHPYH_OKgQqqKZs-WcstcKgpghDlHN8&m=JCSmwF3soIF5i5oJw54jFMkr1w7gJif6iGo27lOImSQI2PDWfh1mxi-m2T3NK7zu&s=MHwkpQKmEstG4YRhT4X1sO_79Qg6_ZnyuQcFjzkfqWw&e=>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

tnelson-integral commented:
Hi @ssolson,
The problem was the result of a wrong style file being used and one that had been overwritten. I updated the style files and all demo in the style files, PacWave, and Tanana to point to the correct style files. Please find the new files on OneDrive. I updated the documentation for the new style file names.

ssolson commented:
Thanks @tnelson-integral I'll check it out tomorrow or Friday.

ssolson commented:
@tnelson-integral 

I have worked on getting the tutorial files to have a consistent snake_case naming as it was the predominate style present in the filenames. I am happy to consider a different style if you prefer as long as we keep consistent naming convention across the files.

As I have renamed most things please ensure you download the latest Tabbed GUI DEMO files before pushing any changes.

Additionally I simplified the localize process so that the python script automatically looks for the tutorial files and style files. Now the user can run the localize script with no input assuming they keep the structure from the dropbox. This will be covered in the tutorials.

I need 1 file from you. In the style files there is a `critical_velocity,<style_folder>\layer_style\velocity_critical_velocity.qml` for which `velocity_critical_velocity.qml` does not exist. Could you create or upload this file for me? Again as I have renamed most things please ensure you download the latest Tabbed GUI DEMO files before making and pushing any changes.

Next week I will update the tutorial sections in the docs to match the new localization and consistent naming format.

ssolson commented:
Hey @tnelson-integral I wanted to check in to see if you had a chance to add the style file mentioned in the previous comment?

ssolson commented:
I see it. Thanks @tnelson-integral. I'll get going through the documentation tutorials again this Friday.

ssolson commented:
@tnelson-integral when I run the tanana river example the critical velocity file you just added shows up as a giant blue square.

Did this work for you when you tested it?

![image](https://github.com/sandialabs/seat-qgis-plugin/assets/13438942/85b53c4a-cd53-4a79-9355-144a7cf110a9)

tnelson-integral commented:
Hi Sterling,
The critical velocity is a constant (single) value so it would show up as a square in this case. It would be variable if a variable GeoTIFF was the input.
One option I could add is to mask the single value to the river [only areas where there is model data]. Though that depends on the model used and if the user wants to the variability along the river (flooding), etc.  Let me know if you would like me to implement the masking.
-Tim

Tim Nelson (he/him/his) | Project Scientist
Tel: 561.240.5096 | Cell: 803.319.3344 | St. Petersburg | FL
***@***.*** | Bio<https://www.integral-corp.com/staff/timothy-r-nelson-ph-d>

<https://www.integral-corp.com/>[cid:integral_1a04a855-2877-4c4f-bd98-b89de3950d0f.png]<https://www.integral-corp.com/>

________________________________
From: ssolson ***@***.***>
Sent: Friday, April 5, 2024 1:09 PM
To: sandialabs/seat-qgis-plugin ***@***.***>
Cc: Tim Nelson ***@***.***>; Mention ***@***.***>
Subject: Re: [sandialabs/seat-qgis-plugin] Revised GUI interface and workflow (PR #31)

[CAUTION: External email. Think before you click links or open attachments.]

@tnelson-integral<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_tnelson-2Dintegral&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=2OT5Kb-Guh1vdHPYH_OKgQqqKZs-WcstcKgpghDlHN8&m=yyEBdH6FSIibnqyLHD431gk0YtoPGWZUwiXyQ5peMRJF74OM2pePQESFicb1S_37&s=AtC40CtZbIdP-dM5X77NjRSfq60vdC4fAaRYQTwRuHE&e=> when I run the tanana river example the critical velocity file you just added shows up as a giant blue square.

Did this work for you when you tested it?

image.png (view on web)<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sandialabs_seat-2Dqgis-2Dplugin_assets_13438942_85b53c4a-2Dcd53-2D4a79-2D9355-2D144a7cf110a9&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=2OT5Kb-Guh1vdHPYH_OKgQqqKZs-WcstcKgpghDlHN8&m=yyEBdH6FSIibnqyLHD431gk0YtoPGWZUwiXyQ5peMRJF74OM2pePQESFicb1S_37&s=lRMyBVvxF_iD8Ro5jO1s52QYVxYZauC-Ll5zByOvink&e=>

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sandialabs_seat-2Dqgis-2Dplugin_pull_31-23issuecomment-2D2040278743&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=2OT5Kb-Guh1vdHPYH_OKgQqqKZs-WcstcKgpghDlHN8&m=yyEBdH6FSIibnqyLHD431gk0YtoPGWZUwiXyQ5peMRJF74OM2pePQESFicb1S_37&s=etdot_Yn6p-phYlmJEuugfYFPVxntrT7YjknYjydEE4&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AZARDBISKAYJC3TH6V55L2LY33LDRAVCNFSM6AAAAABAXB2PBOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDANBQGI3TQNZUGM&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=2OT5Kb-Guh1vdHPYH_OKgQqqKZs-WcstcKgpghDlHN8&m=yyEBdH6FSIibnqyLHD431gk0YtoPGWZUwiXyQ5peMRJF74OM2pePQESFicb1S_37&s=FMb2mfUSBrVXuqQ8gfeiXN__yXtqlU4BACD2cazKjpw&e=>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

ssolson commented:
@tnelson-integral I think the layer needs to be masked because as is this is stating that the critical velocity of dirt, trees, and buildings is  0.05 which I don't think is correct?
![image](https://github.com/sandialabs/seat-qgis-plugin/assets/13438942/0d4695bb-4c9b-4964-9895-2728e0f9bc14)

tnelson-integral commented:
I will work on applying a mask, I will be out of the office most of next week and the following Monday so I might not get to this until 4/16.  However, for clarification, this is the critical velocity for fish larvae and similar as opposed to sediment, so the value is not tied to a specific feature or bed type.
-Tim

Tim Nelson (he/him/his) | Project Scientist
Tel: 561.240.5096 | Cell: 803.319.3344 | St. Petersburg | FL
***@***.*** | Bio<https://www.integral-corp.com/staff/timothy-r-nelson-ph-d>

<https://www.integral-corp.com/>[cid:integral_1a04a855-2877-4c4f-bd98-b89de3950d0f.png]<https://www.integral-corp.com/>

________________________________
From: ssolson ***@***.***>
Sent: Friday, April 5, 2024 1:41 PM
To: sandialabs/seat-qgis-plugin ***@***.***>
Cc: Tim Nelson ***@***.***>; Mention ***@***.***>
Subject: Re: [sandialabs/seat-qgis-plugin] Revised GUI interface and workflow (PR #31)

[CAUTION: External email. Think before you click links or open attachments.]

@tnelson-integral<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_tnelson-2Dintegral&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=2OT5Kb-Guh1vdHPYH_OKgQqqKZs-WcstcKgpghDlHN8&m=fh25jxn5mKiGQCaPlcreJCfsEL8S_M5ZGk_vum72785PC20HiISoZa8qatW96nwU&s=fl4JbddK966xMQy3i_0ok2FKIE7aKY3pDuzA3ZohAnw&e=> I think the layer needs to be masked because as is this is stating that the critical velocity of dirt, trees, and buildings is 0.05 which I don't think is correct?
image.png (view on web)<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sandialabs_seat-2Dqgis-2Dplugin_assets_13438942_0d4695bb-2D4c9b-2D4964-2D9895-2D2728e0f9bc14&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=2OT5Kb-Guh1vdHPYH_OKgQqqKZs-WcstcKgpghDlHN8&m=fh25jxn5mKiGQCaPlcreJCfsEL8S_M5ZGk_vum72785PC20HiISoZa8qatW96nwU&s=TuF2MliqwxIR1vkOxiZmPzhcagI5xrXHo_9zVgxrVKc&e=>

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sandialabs_seat-2Dqgis-2Dplugin_pull_31-23issuecomment-2D2040320499&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=2OT5Kb-Guh1vdHPYH_OKgQqqKZs-WcstcKgpghDlHN8&m=fh25jxn5mKiGQCaPlcreJCfsEL8S_M5ZGk_vum72785PC20HiISoZa8qatW96nwU&s=jmWXbwZ9h4srd5M6WKcJhUO6hVfMN3Lfrcsr2jD8tdE&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AZARDBI52KJS7MSQ7HO673TY33O3TAVCNFSM6AAAAABAXB2PBOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDANBQGMZDANBZHE&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=2OT5Kb-Guh1vdHPYH_OKgQqqKZs-WcstcKgpghDlHN8&m=fh25jxn5mKiGQCaPlcreJCfsEL8S_M5ZGk_vum72785PC20HiISoZa8qatW96nwU&s=HprK2dh4FxBUn7VKUd-sgPpT860iZV6PadioQO8W7xo&e=>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

ssolson commented:
@tnelson-integral I have created https://github.com/IntegralEnvision/seat-qgis-plugin/pull/3 which contain the last updates needed for the Tabbed GUI PR here.

anellenson commented on docs/src/tutorials/prepare_style_files.rst:
Slightly confusing right now. 

What's the correct way to prepare the tutorial files? What are the tutorial files?

1) Style Files
- replace <style_folder> with <style\file\path>
- replace .default to .csv
Can either do manually or with "seat/adjust_file_path.py" in the GitHub module

2) Tutorial Files
 - replace <input_folder> with path/to/input/files
 - replace .default to .ini
 Can either do manually or with "DEMO Files Tabbed GUI/localize_tutorial_files.py" in the DEMO folder 

Seems like we should merge the localize_tutorial_files.py and  "seat/adjust_file_path.py" into one, and put in the proper location (probably with the DEMO files?) 

Also should explain the manual steps for both the style files and input files in one go

ssolson replied:
@anellenson thanks for starting your review.

I deleted this file and you should too. It looks like you made a commit on this file to your local branch before pulling the PR that time merged. Therefore it still exists on your side.
![image](https://github.com/sandialabs/seat-qgis-plugin/assets/13438942/093916d0-24b8-497d-b8f8-e9ae648558be)


If you do the build though you get a warning that this file is not included.
![image](https://github.com/sandialabs/seat-qgis-plugin/assets/13438942/5243a24f-3c86-4766-8881-781c3382540a)

Going forward it will be much more clear process if you could work on your own branch and propose changes to the documentation via a PR.

ssolson commented on docs/src/media/default_files.png:
@anellenson it looks like you accidentally committed the png and the webp file

ssolson commented:
![image](https://github.com/sandialabs/seat-qgis-plugin/assets/13438942/56453cbe-aec2-4674-9e09-69e05416cc58)
🚀🚀🚀 @tnelson-integral Thanks for fixing this! Great work!

tnelson-integral commented:
Sterling,
- The critical velocity receptor is now masked to model data domain only. 
- The style file "velocity_critical_velocity.qml" was updated on the OneDrive so make sure to download the updated stylefile. 
- A bug where an error was generated when closing the plugin window without clicking ok was fixed.

ssolson commented:
> Sterling,
> 
> * The critical velocity receptor is now masked to model data domain only.
> * The style file "velocity_critical_velocity.qml" was updated on the OneDrive so make sure to download the updated stylefile.
> * A bug where an error was generated when closing the plugin window without clicking ok was fixed.

Thanks Tim! Depending on how the week goes (we have a MHKiT release which is could go out this week) I will try out the latest demo files either this week or next..

ssolson commented:
@tnelson-integral apologies on the delay. The MHKiT release took more effort than expected.

This latest update resolves the critical velocity issue and looks good overall. 

I think we just need @anellenson review/ PR and we should be able to close this out.

tnelson-integral commented:
@ssolson This should be ready to be merged to main.

ssolson commented:
@tnelson-integral I think it should be too. I want to build and check it all out one last time. Will follow up today.

ssolson commented:
@tnelson-integral I found one small formatting issue and opened a PR to address.

As soon as you merge that I will merge this PR into main.

tnelson-integral commented:
@ssolson PR merged. Thanks!
